dgram: add UV_UDP_REUSEPORT for udp (#55403)
nodejs/node: doc/api/dgram.md

`exclusive` is set to `false` (the default), cluster workers will use the same
underlying socket handle allowing connection handling duties to be shared.
When `exclusive` is `true`, however, the handle is not shared and attempted

> port sharing results in an error. If creating a `dgram.Socket` with `reusePort`
> flag, `exclusive` will always be `true` when call `socket.bind()` with a port.

Reviewer: ```suggestion
port sharing results in an error. Creating a `dgram.Socket` with the `reusePort`
option set to `true` causes `exclusive` to always be `true` when `socket.bind()`
is called.
```

Author: Thanks! All done !

  * `type` {string} The family of socket. Must be either `'udp4'` or `'udp6'`.
    Required.
  * `reuseAddr` {boolean} When `true` [`socket.bind()`][] will reuse the
    address, even if another process has already bound a socket on it, but
    only one socket can receive the data.
    **Default:** `false`.

>   * `reusePort` {boolean} When `true` [`socket.bind()`][] will reuse the
>     address, even if another process has already bound a socket on it. Incoming
>     datagrams are distributed across the receiving sockets. The flag is available
>     only on some platforms, such as Linux 3.9+, DragonFlyBSD 3.6+, FreeBSD 12.0+,
>     Solaris 11.4, and AIX 7.2.5+. On unsupported platforms this function will
>     return an error.

Reviewer: ```suggestion
  * `reusePort` {boolean} When `true` [`socket.bind()`][] will reuse the
    port, even if another process has already bound a socket on it. Incoming
    datagrams are distributed to listening sockets. The option is available
    only on some platforms, such as Linux 3.9+, DragonFlyBSD 3.6+, FreeBSD 12.0+,
    Solaris 11.4, and AIX 7.2.5+. On unsupported platforms this option raises an
    an error when the socket is bound.
```
